src/analysis/sensitivity.jl

```diff
@@ -401,18 +402,45 @@ ADRIA.sensitivity.rsa(X, y; S=10)
    https://dx.doi.org/10.1002/9780470725184
    Accessible at: http://www.andreasaltelli.eu/file/repository/Primer_Corrected_2022.pdf
 """
-function rsa(X::DataFrame, y::AbstractVector{<:Real}; S::Int64=10)::NamedDimsArray
+function rsa(
+    X::DataFrame, y::AbstractVector{<:Real}, model_spec::DataFrame; S::Int64=10
+)::NamedDimsArray
     N, D = size(X)
-    seq = collect(0.0:(1/S):1.0)
 
     X_di = @MVector zeros(N)
+    sel = trues(N)
+    factors = Symbol.(names(X))
+
+    factors_to_assess = model_spec.fieldname .∈ [factors]
+    foi_attributes = Symbol[:fieldname, :ptype, :lower_bound, :upper_bound]
+    foi_spec = model_spec[factors_to_assess, foi_attributes]
+
+    foi_cat = (foi_spec.ptype .== "categorical")
+    if any(foi_cat)
+        max_bounds = maximum(
+            foi_spec[foi_cat, :upper_bound] .- foi_spec[foi_cat, :lower_bound]
+        )
+        S = round(Int64, max(S, max_bounds))
+    end
```

Comment:
Ideally common/duplicate code that are shared across different functions/methods should be separated out into their own functions.

I'm not going to check this in further detail right now as it looks like it's just copy/pasted from the `outcome_map` method. I'm not sure it will work without any more changes.

Have you checked?

Reply:
As I said, I have checked and would not have submitted a PR if I hadn't. I will group the common code into a function.